src/position/txs/fees-and-rewards.ts

```diff
@@ -34,14 +34,19 @@ export async function buildMultipleCollectFeesAndRewardsTx(
 
   const collectPositionTransactions: TransactionBuilder[] = [];
 
+  // If we don't create an empty map here when resolvedAssociatedTokenAddresses is undefined,
+  // then the ataMap ends up getting set in the buildSingleCollectFeeAndRewardsTx
+  // and not shared across the multiple fee collection txs
+  const ataMap = resolvedAssociatedTokenAddresses ?? {};
   for (const positionAddress of positionAddresses) {
     const txn = await buildSingleCollectFeeAndRewardsTx(
       positionAddress,
       dal,
       client,
       provider,
-      resolvedAssociatedTokenAddresses
+      ataMap
```

Comment:
suggestion: don't make `ataMap` optional in `buildSingleCollectFeeAndRewardsTx` to ensure that something is passed in

Reply:
I was leaving this as is, since the entire logic for HarvestAll should be updated to de-dupe and pack the transactions more succinctly